src/leet_cc_fmt_plug.c

```diff
@@ -212,14 +212,6 @@ static void *get_binary(char *ciphertext)
 	return out;
 }
 
-static int get_hash_0(int index) { return crypt_out[index][0] & PH_MASK_0; }
-static int get_hash_1(int index) { return crypt_out[index][0] & PH_MASK_1; }
```

Comment:
Would it be better to have code for BE that does this (but only for BE)  (NOTE, untested)

```c
static int get_hash_1(int index) { return JOHN_SWAP(crypt_out[index][0]) & PH_MASK_1; }
```

and if leet is a 64 bit format (I am not sure)
```c
static int get_hash_1(int index) { return JOHN_SWAP(crypt_out[index][1]) & PH_MASK_1; }
```

Reply:
The `fmt_default_get_hash` approach seems to be working on both LE and BE systems. Should we just go with it instead of using the BE specific `get_hash_*` functions?

Reply:
Jim, why would the byte swap be better? Do you suggest some of the bytes are always zero? Why would they be?

Reply:
OK. I now see that both `fmt_default_binary_hash` and `fmt_default_get_hash` simply do `return 0`. No wonder why it works so well :-)

Reply:
Also, even if some bytes would be always zero (I don't see that from the code), why would they be in different parts of the 64-bit words depending on endianness? (Your suggestion sounds wrong to me.)

More importantly, though, Dhiru is dropping these hash functions altogether, which will cripple this format's performance for many hashes/salt, and it is totally unclear why that would be fixing any issue. Probably we need to keep the functions and find and fix the actual bug instead, which is probably elsewhere and is still present even after this commit.

Reply:
The is no speed difference between the existing approach, and the `fmt_default_get_hash` approach.

Reply:
No speed difference when you don't have many hashes per salt, sure. There will be when you do.

Reply:
There will also be warnings from the loader:

                                        fprintf(stderr, "Warning: "
                                            "excessive partial hash "
                                            "collisions detected\n%s",
                                            db->password_hash_func !=
                                            fmt_default_binary_hash ? "" :
                                            "(cause: the \"format\" lacks "
                                            "proper binary_hash() function "
                                            "definitions)\n");

Reply:
As far as I can tell, these changes are unrelated to any bug, and Jim is wrong about needing special handling for BE in these functions. Please look for the actual bug elsewhere in the format. I'm sure there are many issues in it. ;-)

Reply:
Also, in a comment in #2414 Dhiru indirectly suggested there's an out of bounds write by this format - so maybe that's the actual bug, or is it a separate bug?